instrumentation/armeria/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaTelemetryBuilder.java

```diff
@@ -215,85 +174,46 @@ public ArmeriaTelemetryBuilder setEmitExperimentalHttpClientMetrics(
   @CanIgnoreReturnValue
   public ArmeriaTelemetryBuilder setEmitExperimentalHttpServerMetrics(
       boolean emitExperimentalHttpServerMetrics) {
-    this.emitExperimentalHttpServerMetrics = emitExperimentalHttpServerMetrics;
+    clientBuilder.setEmitExperimentalHttpClientMetrics(emitExperimentalHttpServerMetrics);
+    serverBuilder.setEmitExperimentalHttpServerMetrics(emitExperimentalHttpServerMetrics);
     return this;
   }
 
   /** Sets custom client {@link SpanNameExtractor} via transform function. */
+  @SuppressWarnings({"rawtypes", "unchecked"})
   @CanIgnoreReturnValue
   public ArmeriaTelemetryBuilder setClientSpanNameExtractor(
       Function<
               SpanNameExtractor<RequestContext>,
```

Comment:
I think this could be changed to `SpanNameExtractor<? super ClientRequestContext>` or `SpanNameExtractor<ClientRequestContext>`  and in the builder you could also change `REQUEST` -> `? super REQUEST`

Reply:
wonderful - I spend some time, but couldn't figure it out